flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBMapState.java

```diff
@@ -520,7 +527,7 @@ private boolean underSameKey(byte[] rawKeyBytes) {
 				return false;
 			}
 
-			for (int i = keyPrefixBytes.length - 1; i >= backend.getKeyGroupPrefixBytes(); --i) {
+			for (int i = keyPrefixBytes.length - 1; --i >= backend.getKeyGroupPrefixBytes(); ) {
```

Comment:
If you want to change like I proposed, I think you need to also remove the `- 1` from the initialization of i. This looks like a bug now.

Reply:
You are right, I have addressed this like your proposed.